### PR TITLE
✨ Source GitHub: add stream `ProjectsV2`

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolve:
- https://github.com/airbytehq/airbyte/issues/25438
- https://github.com/airbytehq/airbyte/issues/27677

## How

- add stream `ProjectsV2`
- limit `page_size` in stream `PullRequestStats` to 10 

## Recommended reading order
1. `streams.py`

## 🚨 User Impact 🚨

no breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`94a033acec`](https://github.com/airbytehq/airbyte/pull/30731/commits/94a033acec473cddc5b58b71d8eab64450524c87)) - ❌

⏲️  Total pipeline duration: 42.01s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Connector package install                                            | ✅        |
| Build source-github docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ❌        |
| Code format checks                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-github/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-github-add-projectsv2/1695668966/94a033acec473cddc5b58b71d8eab64450524c87/source-github/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:94a033acec473cddc5b58b71d8eab64450524c87)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`2baac40391`](https://github.com/airbytehq/airbyte/pull/30731/commits/2baac403917e83e67e857023e7a8a72afffaf82e)) - ✅

⏲️  Total pipeline duration: 05mn45s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Connector package install                                            | ✅        |
| Build source-github docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |
| Code format checks                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-github/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-github-add-projectsv2/1695669881/2baac403917e83e67e857023e7a8a72afffaf82e/source-github/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:2baac403917e83e67e857023e7a8a72afffaf82e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`b28b8b2de5`](https://github.com/airbytehq/airbyte/pull/30731/commits/b28b8b2de57b4f9899bbdee6c9db0a85d4e561bf)) - ✅

⏲️  Total pipeline duration: 06mn44s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Connector package install                                            | ✅        |
| Build source-github docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |
| Code format checks                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-github/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-github-add-projectsv2/1695677243/b28b8b2de57b4f9899bbdee6c9db0a85d4e561bf/source-github/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:b28b8b2de57b4f9899bbdee6c9db0a85d4e561bf)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

File: airbyte-integrations/connectors/source-github/source_github/streams.py [hunk 1]
```diff
@@ -909,6 +916,63 @@ def transform(self, record: MutableMapping[str, Any], stream_slice: Mapping[str,
...
+    http_method = "POST"
+
+    def path(
+        self, *, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "graphql"
+
+    def raise_error_from_response(self, response_json):
+        if "errors" in response_json:
+            raise Exception(str(response_json["errors"]))
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-github/source_github/streams.py:934]: should this be `AirbyteTracedException`?

File: airbyte-integrations/connectors/source-github/source_github/graphql.py [hunk 2]
```diff
@@ -61,6 +61,40 @@ def get_query_pull_requests(owner, name, first, after, direction):
     return str(op)
 
 
+def get_query_projectsV2(owner, name, first, after, direction):
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-github/source_github/graphql.py:64]: let's add some unit tests please

File: airbyte-integrations/connectors/source-github/source_github/github_schema.py
Comment (davydov-d) [inline airbyte-integrations/connectors/source-github/source_github/github_schema.py:1]: what is this giant?

**Review by davydov-d — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-github/source_github/github_schema.py
Comment (artem1205) [reply airbyte-integrations/connectors/source-github/source_github/github_schema.py:1]: we use a sgqlc library for graphql: https://github.com/profusion/sgqlc/tree/master/examples/github

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`cb3458cbd8`](https://github.com/airbytehq/airbyte/pull/30731/commits/cb3458cbd80d4fd150cb32984d222e4c8985b853)) - ✅

⏲️  Total pipeline duration: 05mn42s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Connector package install                                            | ✅        |
| Build source-github docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |
| Code format checks                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-github/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-github-add-projectsv2/1695829721/cb3458cbd80d4fd150cb32984d222e4c8985b853/source-github/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:cb3458cbd80d4fd150cb32984d222e4c8985b853)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

File: airbyte-integrations/connectors/source-github/source_github/streams.py (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-github/source_github/streams.py]: removed; replaced with standard `should_retry`

File: airbyte-integrations/connectors/source-github/source_github/graphql.py (hunk 2 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-github/source_github/graphql.py:64]: added test for query

**Comment by airbyte-oss-build-runner** [comment]
##  source-github test report (commit [`833ce62eba`](https://github.com/airbytehq/airbyte/pull/30731/commits/833ce62eba2d5c358861d71dbe3b1081160e379e)) - ✅

⏲️  Total pipeline duration: 05mn41s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Connector package install                                            | ✅        |
| Build source-github docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |
| Code format checks                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-github/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/artem1205_source-github-add-projectsv2/1695845938/833ce62eba2d5c358861d71dbe3b1081160e379e/source-github/1.3.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:833ce62eba2d5c358861d71dbe3b1081160e379e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-github test
```

**Review by davydov-d — APPROVED** [review]

**Pull request merged**